CONTRIBUTING.md

```diff
@@ -41,42 +41,46 @@ To help us with that, you can run this from the root of your project `cd IHP &&
 
 It is important to update your custom `default.nix` file and set the `rev` to the latest commit every time you perform a `git pull` from within IHP. This is because certain components continue to use the version defined in `default.nix`, even if you have a local IHP.
 
-### Alternative method
+### Running the latest IHP `master`
 
-Another workflow, instead of the simpler `devenv up`, is to use `make console` to load your application together with the framework located in `IHP`. In a `nix-shell`:
+When contributing to IHP core you will want to have your PRs synced with `master`. There is a relationship you'd need to keep with the
+latest [IHP commit](https://github.com/digitallyinduced/ihp/commits/master) and ihp-boilerplate's [flake.lock](https://github.com/digitallyinduced/ihp-boilerplate/blob/master/flake.lock). 
 
-```
-ghci
-$ghci> :l Main
-```
+Your `flake.nix` should have this line:
 
-This will now load your application and all the haskell files in `IHP`.
-
-As we are not using the development tooling for the framework development process we need to manually start the postgres process by running `postgres -D build/db/state -k $PWD/build/db -c "listen_addresses="` in another terminal window.
+```nix
+{
+    ihp.url = "github:digitallyinduced/ihp";
+}
+```
 
-After postgres is started you can now start the application with the local framework version by running:
+Then every time you'd like to update to the latest master, you'll copy/paste the [flake.lock](https://github.com/digitallyinduced/ihp-boilerplate/blob/master/flake.lock) to your own `flake.lock` and run:
 
 ```
-main
+nix flake update
+direnv allow
 ```
 
-After you have made modifications to files inside `IHP`, you need to press `CTRL + C` to stop the process running in `ghci` and then type `:r` to refresh the haskell modules. Now type `main` to start the server again.
+Note that the `flake.lock` on ihp-boilerplate's is rebuilt by IHP's GitHub action whenever a new commit is done on `master`. So be aware that if there's a new commit on master, and its GitHub action isn't done yet, then there will be a mismatch between IHP's commit and the `flake.lock`. You will notice in this case that trying to rebuild IHP takes a really long time, as it will try to compile it, instead of getting it from cachix. In that case, you can wait for the GitHub action to complete or point to a specific IHP commit.
```

Comment:
> or point to a specific IHP commit.

@mpscholten Right now the [flake.lock](https://github.com/digitallyinduced/ihp-boilerplate/blob/master/flake.lock) commit shows the IHP commit message. Would be nice to also include the commit hash.

So it's easier to match an IHP commit and the right ihp-boilerplate `flake.lock`

Reply:
I think the `flake.lock` can be ignored mostly now that the cachix situation is fixed. To update to master you only need to make sure that the url is set to ` ihp.url = "github:digitallyinduced/ihp";` and then run `nix flake update`. The `flake.lock` doesn't need to be copied manually.

Reply:
Yeah, you should generally avoid touching lockfiles manually. We don't even need a pre-defined lockfile in the boilerplate repo, I opened a PR for that: https://github.com/digitallyinduced/ihp-boilerplate/pull/29

Reply:
Even better, closing 👍

Reply:
Also: We include the boilerplate repo itself as input in the IHP flake and expose it as template. So conceivably in the future we could use the Nix-/flake-native `nix flake init github:digitallyinduced/ihp/v1.1` (or any other release or commit) in ihp-new.

If we keep the locked boilerplate commit in sync from the IHP side, it will then use the appropriate boilerplate commit and init a folder from that.

(advanced users could use `nix flake init` directly as well)

Reply:
Actually, maybe we should keep a note about the mismatch of IHP commit, while GitHub action is being built? Because on that phase, cachix didn't cache yet.

Reply:
Updated PR